src/orbit-db-identity-provider.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+const IdentityProvider = require('./identity-provider')
+const Keystore = require('orbit-db-keystore')
+const type = 'orbitdb'
+
+class OrbitDBIdentityProvider extends IdentityProvider {
+  constructor(options = {}) {
+    super()
+    this._keystore = options.keystore || Keystore.create(options.keypath || './orbitdb/identity/keys')
+  }
+
+  // Returns the type of the identity provider
+  static get type () { return type }
+
+  async createId(options = {}) {
+    const ipfs = options.ipfs
+    if (!ipfs)
+      throw new Error('ipfs instance required')
+
+    const keystore = options.keystore || this._keystore
+    const { id } = await ipfs.id()
+    this.id = id
+    const key = await keystore.getKey(id) || await keystore.createKey(id)
+    return key.getPublic('hex')
+  }
+
+  async signIdentity(pubKeyIdSig, options = {}) {
+    const keystore = options.keystore || this._keystore
+    const id = this.id
```

Comment:
I don't think we should keep the `id` in the Provider. The `id` is specific to the *Identity* and not the IdentityProvider. Perhaps we can add `id` to the arguments of the function?